velox/functions/prestosql/types/IPPrefixType.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/type/SimpleFunctionApi.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox {
+
+class IPPrefixType : public VarbinaryType {
```

Comment:
It's a little bit waste of memory to use `VARBINARY` as physical type here: `StringView` needs 16 bytes and the string buffer needs 17 bytes per row, totally 33 bytes per row.
@mbasmanova Is this because the usage of this type is rare and we want to simplify the implementation to trade off some performance?

Reply:
Some more information, I did try to attempt to create the type via an opaque type. However it seems that there is no support for encoding, column reader, and I presume further down. The java type is a fixed width type of 17 bytes.

Reply:
Did you try `ROW(HUGEINT, TINYINT)`?  This seems the most efficient way of storing it.  Also the conversion of `IPPREFIX` -> `IPADDRESS` will be a no cost operation.

Reply:
@Yuhta Thank you for the suggestion. Made change to Row type with overridden arguments method. Please confirm if that is what you were thinking? 

Thanks again!